lib/docker-sync/dependencies.rb

```diff
@@ -0,0 +1,37 @@
+require 'mkmf'
+require 'thor/shell'
+
+Dir[
+  File.join(File.dirname(__FILE__), 'dependencies', 'package_managers', 'base.rb'),
+  File.join(File.dirname(__FILE__), 'dependencies', '**', '*.rb')
+].each { |f| require f }
+
+module DockerSync
+  module Dependencies
+    UNSUPPORTED_OPERATING_SYSTEM = 'Unsupported operating system. Are you sure you need DockerSync?'.freeze
+
+    def self.ensure_all!(config)
+      return ensure_all_for_mac!(config)   if Environment.mac?
+      return ensure_all_for_linux!(config) if Environment.linux?
+      raise(UNSUPPORTED_OPERATING_SYSTEM)
+    end
+
+    def self.ensure_all_for_mac!(config)
+      PackageManager.ensure!
+      Docker.ensure!
+
+      if config.unison_required?
+        Unison.ensure!
+      end
+
+      if config.rsync_required?
+        Rsync.ensure!
+        Fswatch.ensure!
```

Comment:
technically `fswatch` is only needed when `watch_strategy` is `fswatch`.. I'm aware that this conditional is how it currently is in master, though, since we're refactoring.. should we actually split checking for `fswatch` into it's own branch? wdyt?